src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCompilerContext.cs

```diff
@@ -40,5 +40,24 @@ public override FieldLayoutAlgorithm GetLayoutAlgorithmForType(DefType type)
                 return _r2rFieldLayoutAlgorithm;
             }
         }
+
+        public override void SetSystemModule(ModuleDesc systemModule)
+        {
+            base.SetSystemModule(systemModule);
+
+            // System.Object has an EEType field in the standard AOT version used in this repo.
+            // Make sure that we always use the CoreCLR version which (currently) has no fields.
+            var objectType = systemModule.GetType("System", "Object");
+            Debug.Assert(objectType.GetField("m_pEEType") == null, "Incompatible system library. The CoreCLR System.Private.CoreLib must be used when compiling in ready-to-run mode.");
+        }
+
+        /// <summary>
+        /// CoreCLR has no Array`1 type to hang the various generic interfaces off of.
```

Comment:
Nit - perhaps it's just my poor English, is the final "off of" intentional? I would only expect "off" at the end of the sentence.

Reply:
You're correct; it's bad form to end a sentence with a preposition. I don't know what I was thinking of!